### PR TITLE
DOC: Added whatsnew for GH-29174

### PR DESCRIPTION
**Pull request by WillAyd**
follow up to #29174

### CODE REVIEW

**Comment by jbrockmendel** [comment]
LGTM

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -226,6 +226,8 @@ with migrating existing code.
 Removal of prior version deprecations/changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+- Removed the previously deprecated ``raise_with_traceback`` function in ``pandas.compat`` (:issue:`29174`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.0.0.rst]: it's the `assert_raises_regex` that was (semi-)public.

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@WillAyd can you also remove the check for `assert_raises_regex` in `ci\code_checks.sh`

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (WillAyd) [reply doc/source/whatsnew/v1.0.0.rst]: Whoops...nice catch. Will update shortly

**Review by gfyoung — APPROVED** [review]

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -226,6 +226,8 @@ with migrating existing code.
 Removal of prior version deprecations/changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+- Removed the previously deprecated ``assert_raises_regex`` function in ``pandas.util.testing`` (:issue:`29174`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.0.0.rst]: Small nit: the already existing list is 20 lines lower (after "Other removals")

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request merged**